### PR TITLE
New Tab: Pixels

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/0/72649045549333/1207131725936040/f

### Description
Adds pixels to the new NTP
Engagement pixels: https://app.asana.com/0/72649045549333/1207667088727866
Experiment pixels: https://app.asana.com/0/72649045549333/1207667088727867/f

### Steps to test this PR

_When New Tab Page is displayed _
- [x] Open New Tab
- [x] Verify pixel sent `m_new_tab_page_displayed`
- [x] Verify daily pixel with parameters 
- [x] see https://app.asana.com/0/72649045549333/1207667088727866 for rules
- [x] Verify pixel sent `m_new_tab_page_displayed_unique with params: {favorites=1, shortcuts=1, appTP=1, favoriteCount=11_15`

_Clicks on the Edit/Customize button_
- [x] Open New Tab
- [x] Tap on Customize page
- [x] Verify pixel sent `m_new_tab_page_customize`

_Clicks on each shortcut_
- [x] Open New Tab
- [x] Tap on a shortcut
- [x] Verify pixel sent `m_new_tab_page_shortcut_clicked_[shortcut name]`

_Clicks on the info icon_
- [x] Open New Tab
- [x] Ensure you don’t have favorites
- [x] Tap on the icon icon next to placholders
- [x] Verify pixel sent `m_new_tab_page_favorites_info_tooltip`

_When user expands Favourites list_
- [x] Open New Tab
- [x] Ensure you have more than 8 favorites
- [x] Expand list of favorites
- [x] Verify pixel sent `m_new_tab_page_favorites_expanded`

_When user collapses Favourites list_
- [x] Open New Tab
- [x] Ensure you have more than 8 favorites
- [x] With the list expanded, collapse it
- [x] Verify pixel sent `m_new_tab_page_favorites_collapsed`

_When each section is toggled OFF or ON_
- [x] Ensure App Tracking Protection is enabled
- [x] Open New Tab
- [x] Open Customise New Tab
- [x] Disable Favourites section
- [x] Verify pixel sent `m_new_tab_page_customize_section_off_favorites`
- [x] Enable Favourites section
- [x] Verify pixel sent `m_new_tab_page_customize_section_on_favorites`
- [x] Disable Shortcuts section
- [x] Verify pixel sent `m_new_tab_page_customize_section_off_shortcuts`
- [x] Enable Shortcuts section
- [x] Verify pixel sent `m_new_tab_page_customize_section_on_shortcuts`
- [x] Disable AppTP section
- [x] Verify pixel sent `m_new_tab_page_customize_section_off_appTP`
- [x] Enable AppTP section
- [x] Verify pixel sent `m_new_tab_page_customize_section_on_appTP`

_When sections are swapped_
- [x] Open New Tab
- [x] Open Customise New Tab
- [x] Reorder any section
- [x] Verify pixel sent `m_new_tab_page_customize_section_reordered`

_When shortcut enabled / disabled_
- [x] Open New Tab
- [x] Open Customise New Tab
- [x] Enable a shortcut (AI Chat)
- [x] Verify pixel sent `m_new_tab_page_customize_shortcut_added_chat`
- [x] Disable a shortcut (AI Chat)
- [x] Verify pixel sent `m_new_tab_page_customize_shortcut_removed_chat`
- [x] Verify same thing happens with other shortcuts, but with its own name

_When user toggles 'Add Favorite' from Add Bookmark confirmation_
- [x] Navigate to a site
- [x] Add it as bookmark, and toggle Favorite
- [x] Verify pixel sent `m_nav_af_p`
- [x] Verify pixel sent `m_nav_af_p_daily` (should only be sent once per day

_When user clicks on a Favorite on NTP_
- [x] Open New Tab
- [x] Tap on a favourite
- [x] Verify pixel sent `m_favorite_clicked`
- [x] Verify pixel sent `m_favorite_clicked_daily` (should only be sent once per day

_When user clicks on a Bookmark in Bookmark list_
- [x] Open Bookmarks
- [x] Tap on a bookmark
- [x] Verify pixel sent `m_bookmark_launched`
- [x] Verify pixel sent `m_bookmark_launched_daily` (should only be sent once per day

### CODE REVIEW

**Comment by malmstein** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4739?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4739** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4739?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4731** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4731?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#4732](https://github.com/duckduckgo/Android/pull/4732) <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4732?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @malmstein and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: new-tab-page/new-tab-page-impl/src/main/java/com/duckduckgo/newtabpage/impl/pixels/NewTabPixels.kt [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    fun fireShortcutPressed(shortcutName: String)
+    fun fireShortcutAdded(shortcutName: String)
+    fun fireShortcutRemoved(shortcutName: String)
+    fun fireShortcutSectionToggled(enabled: Boolean)
+    fun fireSectionReordered()
+    fun fireNewTabDisplayed()
+}
+
+@ContributesBinding(AppScope::class)
+class RealNewTabPixels @Inject constructor(
```
Comment (marcosholgado) [inline new-tab-page/new-tab-page-impl/src/main/java/com/duckduckgo/newtabpage/impl/pixels/NewTabPixels.kt:45]: tests?

File: new-tab-page/new-tab-page-impl/src/main/java/com/duckduckgo/newtabpage/impl/shortcuts/ShortcutsNewTabSettingsViewModel.kt [hunk 2]
```diff
@@ -60,6 +62,7 @@ class ShortcutsNewTabSettingsViewModel @Inject constructor(
             withContext(dispatchers.main()) {
                 _viewState.update { ViewState(enabled) }
             }
+            pixels.fireShortcutSectionToggled(enabled)
```
Comment (marcosholgado) [inline new-tab-page/new-tab-page-impl/src/main/java/com/duckduckgo/newtabpage/impl/shortcuts/ShortcutsNewTabSettingsViewModel.kt:65]: test that pixel is sent?

**Review by marcosholgado — APPROVED** [review]
LGTM, left a couple of comments, nothing major.

**Comment by malmstein** [comment]
Thanks @marcosholgado !

**Pull request merged**